modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2142,23 +2143,35 @@ def applyier(df, internal_indices, other=[], internal_other_indices=[]):
     # END Map across rows/columns
 
     # __getitem__ methods
+    __getitem_bool = Binary.register(
+        lambda df, r: df[r], join_type="left", labels="drop"
```

Comment:
Why are we doing a left join here? Are we not just applying the boolean mask on all of the partitions and concat-ing those?

Reply:
This is "index join", not "dataframe join". `left` here means "take all items from `df.index` regardless of their presence in `r` and augment them with items from `r.index` which were missing in `df.index`".